### DISCUSSION TITLE
End of macOS support

### DISCUSSION THREAD

**Discussion by foldynl** [foldynl/QLog#385] [General]
QLog has long been released for Linux, Windows, and macOS. However, I am currently only able to support Linux and Windows. macOS support for QLog was initially added by volunteers, but unfortunately, there is no active developer community for macOS anymore. As a result, several issues have been registered that I am unable to solve without external help, including:

  - A black online map
  - Inability to switch to macOS 14 - binaries must be signed and other issues
  - Various strange behaviors

I regret to inform you I will have to discontinue support for this platform. The plan is that no new release will be released for this platform - no DMG packages will be released. However, I still plan to keep code review active so that it will still be compilable under macOS.

Thank you for your understanding.

Regards
Ladislav

**Comment by 1015onasaturdaynight** [comment]
Is there demand for a (signed) macOS version?

**Reply by foldynl** [reply]
may I ask you. The code is signed but what about the online map? Is it black or you can see the online map? Is it prepared for MacOS14 (Apple Silicon) or for Intel?

**Reply by 1015onasaturdaynight** [reply]
I'm still on older macOS x86 as $dayjob is enterprise software development on Linux.

macOS 12.7.5 (Intel) in VMWare Fusion Pro 11.5.7
QLog ac7295d8f30ffd8c437b92075db0f5b5082b9695
Debug build
Qt Creator 13.0.2
qt 6.7.0_1
hamlib 4.5.5
qtkeychain 0.14.3
dbus-glib 0.112
brotli 1.1.0

Here's what it looks like for me:

<img width="1792" alt="Screen Shot 2024-06-14 at 4 09 22 PM" src="https://github.com/foldynl/QLog/assets/172004313/ea6d4314-86e1-4432-8b0e-268c4d1b96d1">

**Reply by aa5sh** [reply]
Ladislav, I just built both on an Intel and Silicon Mac and on both the Maps seems ok to me in just quick testing.  I have not tried to sign the app.  I just launched from QT-Creator and ran the corresponding qlog.app 
![image](https://github.com/user-attachments/assets/22ebd647-ec90-441e-babc-f77dc31ff575)

**Reply by foldynl** [reply]
it looks good. Did you have to make any modifications to the code? Some issue with the Map and the start of the application were reported from users (more details #269 ) but I see that everything looks good.

**Comment by aa5sh** [comment]
I just built the latest version.  I just renewed my Apple Developer account
so I can try to sign the app and play with it some more on other devices.

**Reply by foldynl** [reply]
Excellent . What is your next step? Do you plan to publish the signed DMG?

**Comment by aa5sh** [comment]
I’d be more than happy to do that if it would help you/others.  I will read up on how to properly sign and package a DMG file.  I’ve been at a work conference the last few days so take a loot at it today.  Most of my development the last few years is apps I built in Xojo and it will automatically sign the apps as part of its build process.

I do need to get acquainted with Git.  I made a couple tweaks in code for you to consider.

Michael

**Comment by aa5sh** [comment]
Just wanted to give you an update.  I think I have figured out the signing
and all.

I've been playing with the Maps (Both the Widget and grid in Statistics
window) not showing.  It is odd.  If I follow similar build steps as on the
GitHub Actions when the App is first built it works fine.  Once I run the
macdeployqt step it breaks the map working.  Now that I can reproduce it I
will try and determine what is causing the problem.  May just need a
different switch or something when that is run.


On Jul 26, 2024, at 9:46 AM, Michael Morgan ***@***.***> wrote:

I’d be more than happy to do that if it would help you/others.  I will read
up on how to properly sign and package a DMG file.  I’ve been at a work
conference the last few days so take a loot at it today.  Most of my
development the last few years is apps I built in Xojo and it will
automatically sign the apps as part of its build process.

I do need to get acquainted with Git.  I made a couple tweaks in code for
you to consider.

Michael

On Jul 26, 2024, at 12:35 AM, Ladislav ***@***.***> wrote:


Excellent . What is your next step? Do you plan to publish the signed DMG?

—
Reply to this email directly, view it on GitHub
<https://github.com/foldynl/QLog/discussions/385#discussioncomment-10155677>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AUEEMXX4EBJZMAL4TNTXXATZOHNYTAVCNFSM6AAAAABI2MAH4SVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJVGU3DONY>
.
You are receiving this because you commented.Message ID:
***@***.***>

**Comment by aa5sh** [comment]
Well a little more info.  It looks like some dylib’s are not being copied or referenced right in regards to QtWebEngineCore.  It is throwing several of these errors at runtime after running the macdeployqt.  

dyld[8054]: Library not loaded: @executable_path/../Frameworks/libbrotlidec.1.dylib
  Referenced from: <62AFE231-56DE-354D-81BC-7E191D81DBDA> /Users/michaelmorgan/Documents/GitHub/QLog/build/qlog.app/Contents/Frameworks/QtNetwork.framework/Versions/A/QtNetwork
  Reason: tried: '/Users/michaelmorgan/Documents/GitHub/QLog/build/qlog.app/Contents/Frameworks/QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/Frameworks/libbrotlidec.1.dylib' (no such file)Library not loaded: @executable_path/../Frameworks/libicui18n.74.dylib
  Referenced from: <A78169A1-6B64-33C2-A157-AEA085F7A49A> /Users/michaelmorgan/Documents/GitHub/QLog/build/qlog.app/Contents/Frameworks/QtCore.framework/Versions/A/QtCore
  Reason: tried: '/Users/michaelmorgan/Documents/GitHub/QLog/build/qlog.app/Contents/Frameworks/QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/Frameworks/libicui18n.74.dylib' (no such file)Library not loaded: @executable_path/../Frameworks/libglib-2.0.0.dylib
  Referenced from: <0D74653E-184D-34B4-A10C-2CFA621C334F> /Users/michaelmorgan/Documents/GitHub/QLog/build/qlog.app/Contents/Frameworks/QtGui.framework/Versions/A/QtGui
  Reason: tried: '/Users/michaelmorgan/Documents/GitHub/QLog/build/qlog.app/Contents/Frameworks/QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/Frameworks/libglib-2.0.0.dylib' (no such file)

**Comment by foldynl** [comment]
that's exactly what I haven't been able to test. I used a procedure that someone once sent me. It is possible that something has changed over time.

**Comment by aa5sh** [comment]
I noticed in the github actions that you were copying some files.  I tried
adding this missing dylib as well but got the same result.  I think it is
just something that may need to be be done with a install_name_tool command
that I haven't gotten worked out yet.  Just for future reference (for me
mainly because I forget things) I opened the app from the command line './
qlog.app/Contents/MacOS/qlog'.  When I googled the messages it said the
QtWebEngineView/Widgets can be problematic on MacOS.

**Comment by aa5sh** [comment]
I think I got the issue figured out.  In the action (I was looking at macOSBuild.yml) I changed:
macdeployqt qlog.app
To
macdeployqt qlog.app -executable=./qlog.app/Contents/MacOS/qlog

I also just removed all of the copy commands other commands and it seems ok.  Not sure why they were there.  So my build step is just cd build then macdeployqt qlog.app -executable=./qlog.app/Contents/MacOS/qlog and that’s it and it downloaded and ran fine.

I will upload a dmg to my fork if you want.  But I think if you adjust that build step your’s should be good.  It works ok without signing at this point.

Michael

**Reply by aa5sh** [reply]
I've gotten in my fork to create a signed application that the maps do seem to work.  I haven't fully tested everything on the app.

If someone else wants to try it:  https://github.com/aa5sh/QLog 

![image](https://github.com/user-attachments/assets/dc894d1d-1b91-4ea3-aeb0-782f0bee5385)

**Reply by foldynl** [reply]
I am not able to test it, but I noticed an issue with icons from Resources on your screenshot. Otherwise it is a good news that it is again working under MacOS

![MacOSImg2](https://github.com/user-attachments/assets/c2e05a72-e049-4f09-8c5a-374a685ae123)

**Comment by aa5sh** [comment]
Ok they are displaying now.  For some reason it is needed to run the
macdeployqt twice.  Once with the -executable flag and once without it.
But now the icons are showing:
[image: image.png]

I did load an updated copy of the dmg in my fork.
